server/src/client_api.rs

```diff
@@ -51,6 +51,51 @@ pub async fn get_features(
             .ok_or(EdgeError::ClientFeaturesFetchError(FeatureError::Retriable)),
     }
 }
+#[utoipa::path(
+    context_path = "/api",
+    params(("feature_name" = String, Path,)),    
+    responses(
+        (status = 200, description = "Return feature toggles for this token", body = ClientFeatures),
+        (status = 403, description = "Was not allowed to access feature"),
+        (status = 400, description = "Invalid parameters used"),
+        (status = 404, description = "Feature did not exist in accessible")
+    ),
+    security(
+        ("Authorization" = [])
+    )
+)]
+#[get("/client/features/{feature_name}")]
+pub async fn get_feature(
+    edge_token: EdgeToken,
+    features_cache: Data<DashMap<String, ClientFeatures>>,
+    token_cache: Data<DashMap<String, EdgeToken>>,
+    feature_name: web::Path<String>,
+    req: HttpRequest,
+) -> EdgeJsonResult<ClientFeature> {
+    let validated_token = token_cache
+        .get(&edge_token.token)
+        .map(|e| e.value().clone())
+        .ok_or(EdgeError::AuthorizationDenied)?;
```

Comment:
Do we need this check? Seems to be the same one as in `get_features`. Would be nice if it was handled through a middleware / helper method for DRY-ness.

Reply:
I don't disagree, I'd like that. Haven't been able to work out how to do it without doing what feels like too much logic in a FromRequest implementation. In a middleware you can't put it as an argument to the function in an easy way. The one time where types are fighting us. :) The clue is to get the project(s) and environment that we've fetched from the upstream validate call.

I'm open to suggestions for a better way to do this, we basically do the same in all endpoints.